src/api.v2/model/experiment.js

```diff
@@ -1,22 +1,91 @@
+const _ = require('lodash');
+
+/* eslint-disable func-names */
 const generateBasicModelFunctions = require('../helpers/generateBasicModelFunctions');
+const sqlClient = require('../../sql/sqlClient');
+const { aggregateIntoJson } = require('../../sql/helpers');
+
+const getLogger = require('../../utils/getLogger');
+
+const logger = getLogger('[ExperimentModel] - ');
 
-const tableName = 'experiment';
+const experimentTable = 'experiment';
+const experimentExecutionTable = 'experiment_execution';
 
-const selectableProps = [
+const experimentFields = [
   'id',
   'name',
   'description',
-  'processing_config',
+  'samples_order',
   'notify_by_email',
   'created_at',
   'updated_at',
 ];
 
 const basicModelFunctions = generateBasicModelFunctions({
-  tableName,
-  selectableProps,
+  tableName: experimentTable,
+  selectableProps: experimentFields,
 });
 
+const getExperimentData = async (experimentId) => {
+  const sql = sqlClient.get();
+
+  function query() {
+    this.select('*')
+      .from(experimentTable)
+      .leftJoin(experimentExecutionTable, `${experimentTable}.id`, `${experimentExecutionTable}.experiment_id`)
+      .where('id', experimentId)
+      .as('experiment_with_exec');
+  }
+
+  const experimentExecutionFields = [
+    'params_hash', 'state_machine_arn', 'execution_arn',
+  ];
+
+  const result = await aggregateIntoJson(query, experimentFields, experimentExecutionFields, 'pipeline_type', 'pipelines', sql);
```

Comment:
I'm a bit concerned that this `aggregateIntoJson` is complex enough that other people will avoid using it. It's also kind of hard to read. I've read the documentation and it's still kind of confusing, specially considering that in this case what it needs to do is pretty simple. Do you think this complexity is worth it? We just have 2 extra fields to add. You could only query for the experiment info in the first query and then do
```
experimentInfo = await ....

executions = await this.select('*')
      .from(experimentExecutionTable)
      .where('experiment_id', experimentId)

executions.forEach((exc) => {
    experimentInfo[exc.pipeline_type] = exc
}

```